bank_lib/src/bank.rs

```diff
@@ -0,0 +1,315 @@
+use crate::errors::BankError;
+use crate::operations::Operation;
+use std::collections::{HashMap, VecDeque};
+
+#[derive(Debug)]
+pub struct Bank {
+    accounts: HashMap<String, i64>,
+    history: Vec<Operation>,
+    account_operations: HashMap<String, VecDeque<usize>>,
+}
+
+impl Default for Bank {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl Bank {
+    /// Создает новый экземпляр банка.
+    ///
+    /// # Пример
+    ///
+    /// ```
+    /// use bank_lib::Bank;
+    /// let bank = Bank::new();
+    /// ```
+    pub fn new() -> Self {
+        Bank {
+            accounts: HashMap::new(),
+            history: Vec::new(),
+            account_operations: HashMap::new(),
+        }
+    }
+
+    /// Создает новый счет.
+    ///
+    /// # Пример
+    ///
+    /// ```
+    /// use bank_lib::Bank;
+    /// let mut bank = Bank::new();
+    /// assert!(bank.create_account("Alice".to_string()).is_ok());
+    /// ```
+    ///
+    /// # Ошибки
+    ///
+    /// Возвращает `BankError::AccountExists`, если счет с таким именем уже существует.
```

Comment:
```suggestion
    /// Возвращает [`BankError::AccountExists`], если счет с таким именем уже существует.
```

Если делать красиво, то до конца) Это создает док-ссылку на нужный вариант ошибки

Reply:
Аналогично в других местах можно